mitodl/open-discussions
# add subreddit title to subreddit API and frontend

#### What are the relevant tickets?

closes #257 

#### What's this PR do?

Basically we want to show the channel title in the little byline, not the channel name (which basically a url component). To do this I made a small change to the channel serializer. This basically added a new request to a ton of tests in `channels/views_tests.py` so I had to regenerate a whole bunch of cassettes. As a nice side effect, several tests are now converted to use the new, reproducible factory strategy for creating reddit objects that Nathan came up with. So this should be much less painful for the next person who has to fiddle with it, haha.

#### How should this be manually tested?

When you go to look at the frontpage in the UI (at `/`) you should see the channel title instead of the channel name in the byline. Clicking on the channel title should take you to the post listing for the channel itself. Other than that the code should be clean looking and all the tests should be passing.
